src/test/ui/feature-gates/feature-gate-unsafe_block_in_unsafe_fn.rs

```diff
@@ -0,0 +1,11 @@
+#![deny(unused_unsafe)]
```

Comment:
The behavior of this test should be the same with and without the feature flag, could you ensure that with a test?

Generally I am not sure if there is precedent for such a lint-only feature flag -- as long as `unsafe_op_in_unsafe_fn` is allow-by-default, it should not change behavior, and lint as not subject to the same strict stability guarantees as the language itself. But that is up to @rust-lang/lang.

Reply:
For the moment, I left every change in this PR behind the feature gate. I’d expect that we remove the gate when the lint is fully ready, e.g proposes a suggestion to add an unsafe block for the whole unsafe function. When this happens, this test doesn’t serve any purpose, I added it to make tidy happy :D

Reply:
AFAIK the point of the feature-gate tests is to ensure that you get an error when using a feature without enabling the gate...

Also, that does not resolve my concern that even with the feature flag, this test here behave the same.

Reply:
I don't really see how I could feature-gate this. I think I could either not feature-gate at all, or gate the lint to prevent changing the lint level without the feature gate.

Reply:
Yeah that's why I said, I never saw a feature gate for a lint...

Reply:
We can emit a feature gate error if the lint is enabled without the feature gate being enabled, I think -- it won't stop people `#[allow(...)]` the lint but that seems not horrible.